src/main/scala/com/johnsnowlabs/nlp/annotators/NormalizerModel.scala

```diff
@@ -0,0 +1,78 @@
+package com.johnsnowlabs.nlp.annotators
+
+import com.johnsnowlabs.nlp.{Annotation, AnnotatorModel, ParamsAndFeaturesReadable}
+import com.johnsnowlabs.nlp.AnnotatorType.TOKEN
+import com.johnsnowlabs.nlp.pretrained.ResourceDownloader
+import com.johnsnowlabs.nlp.serialization.MapFeature
+import org.apache.spark.ml.param.{BooleanParam, StringArrayParam}
+import org.apache.spark.ml.util.Identifiable
+
+class NormalizerModel(override val uid: String) extends AnnotatorModel[NormalizerModel] {
+
+  override val annotatorType: AnnotatorType = TOKEN
+
+  override val requiredAnnotatorTypes: Array[AnnotatorType] = Array(TOKEN)
+
+  val pattern = new StringArrayParam(this, "pattern",
+    "normalization regex pattern which match will be replaced with a space")
+
+  val lowercase = new BooleanParam(this, "lowercase", "whether to convert strings to lowercase")
+
+  protected val slangDict: MapFeature[String, String] = new MapFeature(this, "slangDict")
+
+  def this() = this(Identifiable.randomUID("NORMALIZER"))
+
+  def setPattern(value: Array[String]): this.type = set(pattern, value)
+
+  def setLowerCase(value: Boolean): this.type = set(lowercase, value)
+
+  def setSlangDict(value: Map[String, String]): this.type = set(slangDict, value)
+
+  def applyRegexPatterns(word: String): String ={
+
+    val nToken = {
+      get(pattern).map(_.foldLeft(word)((currentText, compositeToken) => {
+        currentText.replaceAll(compositeToken, "")
+      })).getOrElse(word)
+    }
+    nToken
+  }
+
+  protected def getSlangDict: Map[String, String] = $$(slangDict)
+
+  /** ToDo: Review implementation, Current implementation generates spaces between non-words, potentially breaking tokens */
+  override def annotate(annotations: Seq[Annotation]): Seq[Annotation] =
+
+    annotations.map { token =>
+
+      val cased =
+        if ($(lowercase)) token.result.toLowerCase
+        else token.result
+
+      val correctedWords =
+        if ($$(slangDict).contains(cased)) {
+          $$(slangDict)(cased)
+        } else {
+          cased
+        }
+
+      val finalWords = correctedWords.split(" ").map(word => applyRegexPatterns(word))
+
+      Annotation(
+        annotatorType,
+        token.begin,
+        token.end,
+        finalWords.mkString(" "),
+        token.metadata
+      )
+    }.filter(_.result.nonEmpty)
+
```

Comment:
Breaks the abc rule of one annotation per token. Change to map/flatMap finalWords token into annotations